### PR TITLE
Feat: add address details

### PR DESCRIPTION
**Pull request by kevbarns**
- update schemas & types to add full address details (to be renamed **address** and update business logic later)
- update UserRecruteur Collection with **address_detail** 
- update Formulaire Collection with **address_detail** for CFA partners (without user)

### CODE REVIEW

File: server/src/jobs/lba_recruteur/formulaire/misc/updateAddressDetailOnFormulaireCollection.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+runScript(async () => {
+  logger.info("Start update user adresse detail")
+  const formulaires = await Formulaire.find({ adresse_detail: { $eq: null } })
+
+  logger.info(`${formulaires.length} entries to update...`)
+
+  if (!formulaires.length) return
+
+  await asyncForEach(formulaires, async (formulaire, index) => {
+    console.log(`${index}/${formulaires.length} - ${formulaire.gestionnaire ? "delegate" : "entreprise"}`)
```
Comment (rimiti) [inline server/src/jobs/lba_recruteur/formulaire/misc/updateAddressDetailOnFormulaireCollection.ts]:
```suggestion
    logger.info(`${index}/${formulaires.length} - ${formulaire.gestionnaire ? "delegate" : "entreprise"}`)
```

File: server/src/common/utils/asyncUtils.ts [hunk 2]
```diff
@@ -3,3 +3,5 @@ export const asyncForEach = async (array, callback) => {
     await callback(array[index], index, array)
   }
 }
+
+export const delay = (ms) => new Promise((resolve) => setTimeout(resolve, ms))
```
Comment (rimiti) [inline server/src/common/utils/asyncUtils.ts]: Node v18 handle async setTimeout is you want:
https://nodejs.org/docs/latest-v18.x/api/timers.html#settimeoutcallback-delay-args

File: server/src/jobs/lba_recruteur/formulaire/misc/updateAddressDetailOnUserCollection.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          const { errors } = error.response.data
+
+          if (errors.length) {
+            if (
+              errors.includes("Le numéro de siret n'est pas correctement formatté") ||
+              errors.includes("Le siret ou siren indiqué n'existe pas, n'est pas connu ou ne comporte aucune information pour cet appel")
+            ) {
+              console.log(`Invalid siret DELETED : ${user.siret} - User & Formulaire removed`)
+              await UserRecruteur.findByIdAndDelete(user._id)
+              await Formulaire.findOneAndRemove({ id_form: user.id_form })
```
Comment (rimiti) [inline server/src/jobs/lba_recruteur/formulaire/misc/updateAddressDetailOnUserCollection.ts]:
```suggestion
              await Promise.all([UserRecruteur.findByIdAndDelete(user._id), Formulaire.findOneAndRemove({ id_form: user.id_form })]);
```

**Review by rimiti — APPROVED** [review]

File: server/src/common/utils/asyncUtils.ts (hunk 2 above)
Comment (kevbarns) [reply server/src/common/utils/asyncUtils.ts]: good to know 💯 , but still experimental, we'll keep the utils for now

**Pull request merged**